Add "Presentations" tab

Sheargrub opened the pull request:
Adds a new "Presentations" tab, which will pull from the "ciroh_portal_presentation" keyword on HydroShare. This provides a clear location to surface CIROH-affiliated workshops, conference keynotes, and other presentation materials. Also includes a matching submission tab in the "Contribute" page.

Resources that contain the "ciroh_portal_presentation" tag will be excluded from the "Datasets" page unless they are explicitly tagged with "ciroh_portal_data".

I'm hoping to start getting some DevCon workshops uploaded to get things started soon, but I want to make sure that this approach is fully approved first.
 
---

TODO: Establish a curated collection for presentations. (Currently pulls from the dataset page's curated collection.)

---

This PR also bundles in some miscellaneous refactoring/upkeep, including:
- Improved handling for missing API keys
- A new navbar dropdown to highlight both GitHub repositories corresponding to CIROH Portal
- Fixes for dataset tag detection

romer8 commented on docusaurus.config.js:
the dummy is good though to start the `docusuarus`

romer8 commented on src/components/Contribute/PublicationsContribute.js:
good renaming

Sheargrub commented on src/components/Presentations/index.js:
```diff
@@ -0,0 +1,165 @@
+          docs_url: ""
+        }));
+        setResources(mappedList);
+        
```
Thanks for this catch - it ended up leading me to the approach that was used for presentation embedding. In any case, fixed and optimized with the rewritten `mapWithCustomMetaData` function.

Sheargrub commented:
Now ready for review! @romer8

Functional changes:
- New functionality to embed presentation PDFs within the list view
   - Note that tile view couldn't be supported; as such, tile view instead directly links to the presentation
   - This should be generalizable to other HydroShare tabs, though it may need tweaks to support other formats
- Curated tab deprecated
  - The underlying functionality is still used to manually include older HydroShare resources
- HydroShare file uploads now display visual feedback for which items are selected
- HydroShare abstracts now enforce a 150-character minimum instead of a 150-word minimum (per HydroShare guidelines)
- Fixed Contribute page title ("Applications" -> "Contribute")

Technical changes:
- HydroShare file uploading adjusted to use standard-compliant API calls
- HydroShare metadata retrieval fully refactored
- Deleted vestigal `HydroShareResources` component folder, which has been fully replaced

romer8 commented on src/components/HydroShareResourcesRows/styles.module.css:
```diff
@@ -238,6 +238,32 @@
+  width: 100%;
+  height: 100%;
+}
+
```
I added the following for a close button:
```

.closeButton {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
  background: transparent;
  border: none;
  font-size: 2rem;
  cursor: pointer;
  line-height: 1;
  color: var(--ifm-color-secondary);
}

.closeButton:hover,
.closeButton:focus-visible {
  color: var(--ifm-color-primary);
}

```

Sheargrub replied:
Adding shortly!

romer8 commented on src/components/HydroShareResourcesRows/HydroShareResourceRow.js:
I have noticed that the behavior was not consistent between browser (chrome based browsers will download the file, but not show it) I have make changes to make it work with most browsers. I am copying the whole file since it is easier:
```
import React, { useEffect, useState,useRef } from 'react';
import Link from '@docusaurus/Link';
import { MdDriveFileMove } from 'react-icons/md';
import { LiaExternalLinkSquareAltSolid } from 'react-icons/lia';
import { FaBookmark } from "react-icons/fa";
import { IoTvOutline } from "react-icons/io5";
import clsx from 'clsx';
import styles from './styles.module.css';

export default function HydroShareResourceRow({ resource, defaultImage }) {

  const [showEmbed, setShowEmbed] = useState(false);
  const [embedSrc, setEmbedSrc] = useState(null);
  const objectUrlRef = useRef(null);

  const toggleShowEmbed = (e) => {
    e.preventDefault();
    setShowEmbed(!showEmbed);
  }
  const closeEmbed = () => setShowEmbed(false);

  const {
    resource_id,
    title = 'Untitled',
    description,
    thumbnail_url,
    page_url,
    docs_url,
    resource_url,
    embed_url,
  } = resource;
  useEffect(() => {
    // 1. If the embed is requested, fetch the PDF and build a blob URL
    if (showEmbed && embed_url) {
      fetch(embed_url)
        .then(r => r.blob())
        .then(blob => {
          // Revoke any previous URL before replacing it
          if (objectUrlRef.current) {
            URL.revokeObjectURL(objectUrlRef.current);
          }
          const url = URL.createObjectURL(blob);
          objectUrlRef.current = url;
          setEmbedSrc(url);
        });
    } else {
      // 2. If the panel is being hidden, clear the URL immediately
      if (objectUrlRef.current) {
        URL.revokeObjectURL(objectUrlRef.current);
        objectUrlRef.current = null;
      }
      setEmbedSrc(null);
    }

    // 3. Component-unmount (or dependency-change) cleanup
    return () => {
      if (objectUrlRef.current) {
        URL.revokeObjectURL(objectUrlRef.current);
        objectUrlRef.current = null;
      }
    };
  }, [showEmbed, embed_url]);
  return (
    <div id={resource_id} className={styles.columnItem}>
      {/* Unlike tiles, the overarching structure is wrapped in a column to support embeds */}
      <div id={resource_id+"_content"} className={clsx(styles.rowItem, 'card')}>
        {/* ─── FULL-CARD OVERLAY ─── */}
        {/* Covers the whole card so any click that isn't on a "real" link            */}
        {/* triggers the hash–scroll. Not nested: it's a sibling in the DOM.          */}
        <Link
          to={`#${resource_id}`}
          className={styles.fullCardLink}
          title="Scroll to this resource"
        />
        {/* ─── THUMBNAIL ─── */}
        <div className={styles.imageWrapper}>
          {thumbnail_url ? (
            <img src={thumbnail_url} alt={title} className={styles.image} />
          ) : defaultImage ? (
            <img src={defaultImage} alt={title} className={styles.image} />
          ) : (
            <div className={clsx(styles.imagePlaceholder, styles.placeholder)} />
          )}

          <div className={styles.hoverOverlay}>
            <h5 className={styles.overlayTitle}>{title}</h5>
            <div className={styles.overlayIcons}>
              {page_url && (
                <a
                  href={page_url}
                  target="_blank"
                  rel="noreferrer"
                  className={styles.iconLink}
                  title="Website"
                >
                  <LiaExternalLinkSquareAltSolid size={30} />
                </a>
              )}
              {docs_url && (
                <a
                  href={docs_url}
                  target="_blank"
                  rel="noreferrer"
                  className={styles.iconLink}
                  title="Docs Page"
                >
                  <FaBookmark  size={20} />
                </a>
              )}
              {resource_url && (
                <a
                  href={resource_url}
                  target="_blank"
                  rel="noreferrer"
                  className={styles.iconLink}
                  title="Resource Page"
                >
                  <MdDriveFileMove size={30} />
                </a>
              )}
              {embed_url && (
                  <a
                    href="#"
                    role="button"           
                    tabIndex={0}
                    onClick={toggleShowEmbed}
                    onKeyDown={e => {
                      if (e.key === 'Enter' || e.key === ' ') {
                        e.preventDefault();
                        toggleShowEmbed(e);
                      }
                    }}
                    className={styles.iconLink}
                    title={showEmbed ? 'Hide PDF' : 'Display PDF'}
                  >
                  <IoTvOutline size={30} />

                  </a>
              )}
            </div>
          </div>
        </div>

        {/* ─── TEXT BLOCK ─── */}
        <div className={styles.textWrapper}>
          {title ? (
            <h1 className={styles.title}>
              {/* External link → page_url */}
              <a
                href={page_url || resource_url || '#'}
                target="_blank"
                rel="noreferrer"
                className={styles.pageLink}
              >
                {title}
              </a>
            </h1>
          ) : (
            <h1 className={styles.titlePlaceholder} />
          )}

          {description ? (
            <p className={styles.description}>{description}</p>
          ) : (
            <p className={styles.descriptionPlaceholder} />
          )}
        </div>
      </div>
      {showEmbed && (
        <div id={`${resource_id}_embed`} className={clsx(styles.rowItem, 'card')}>
          <button
            type="button"
            onClick={closeEmbed}
            aria-label="Close PDF"
            className={styles.closeButton}
          >
            ×
          </button>
          <div className={styles.embedCentering}>
            <div className={styles.embedWrapper}>
              {embedSrc && (
                <embed
                  src={embedSrc}
                  type="application/pdf"
                  className={styles.embedContent}
                />
              )}
            </div>
          </div>
        </div>
      )}
    </div>
  );
}

```

Sheargrub replied:
Agh, figures that it would break in Chrome after fixing it for Firefox >_>
In any case, thank you! Will merge this shortly after testing
